### PR TITLE
Add POST /api/articles/:article_id/comments model and controller with passing 201, 400 and 404 status tests

### PR DESCRIPTION
**Pull request by davidptclark**
Adds: 
- POST /api/articles/:article_id/comments model and controller
- checkUserExists model
- specific errors for unregistered user, invalid keys in object body, missing fields/keys in object body
- Passing tests for 201, 400 and 404 statuses

### CODE REVIEW

File: controllers/comments.controller.js [hunk 1]
```diff
@@ -11,3 +15,20 @@ exports.getCommentsByArticleId = (req, res, next) => {
       next(err);
     });
 };
+
+exports.postCommentsByArticleId = (req, res, next) => {
+  const { article_id: id } = req.params;
+  const { username: user } = req.body;
+  const body = req.body;
+  Promise.all([
+    checkArticleExists(id),
```
Comment (daviidbartlett) [inline controllers/comments.controller.js:24]: you don't need to check article exists

File: controllers/comments.controller.js [hunk 2]
```diff
@@ -11,3 +15,20 @@ exports.getCommentsByArticleId = (req, res, next) => {
...
+    checkUserExists(user),
```
Comment (daviidbartlett) [inline controllers/comments.controller.js:25]: you don't need to check user exists

**Review by daviidbartlett — COMMENTED** [review]
Please refactor to remove unnecessary models

**Comment by davidptclark** [comment]
The most recent commit takes in David's suggestions to remove functions and use err.detail property to inform PSQL errors

**Pull request merged**